lib/writers/s3.go

```diff
@@ -24,20 +29,35 @@ type S3Writer struct {
 func NewS3Writer(logger logr.Logger, stateStoreSpec platformv1alpha1.BucketStateStoreSpec, destination platformv1alpha1.Destination, creds map[string][]byte) (StateStoreWriter, error) {
 	endpoint := stateStoreSpec.Endpoint
 
-	accessKeyID, ok := creds["accessKeyID"]
-	if !ok {
-		return nil, fmt.Errorf("accessKeyID not found in secret %s/%s", destination.Namespace, stateStoreSpec.SecretRef.Name)
+	opts := &minio.Options{
+		Secure: !stateStoreSpec.Insecure,
 	}
 
-	secretAccessKey, ok := creds["secretAccessKey"]
-	if !ok {
-		return nil, fmt.Errorf("secretAccessKey not found in secret %s/%s", destination.Namespace, stateStoreSpec.SecretRef.Name)
+	logger.Info("setting up s3 client", "authMethod", stateStoreSpec.AuthMethod, "endpoint", endpoint, "insecure", stateStoreSpec.Insecure)
+	switch stateStoreSpec.AuthMethod {
+	case AuthMethodIAM:
+		opts.Creds = credentials.NewIAM("")
+
+	case "", AuthMethodAccessKey: //used to be optional so lets handle empty as the default
```

Comment:
alternatively, we could separate and use `fallthrough`:

```go
case "":
  // log?
  fallthrough
case AuthMethodAccessKey:
  // ...
```

Reply:
also, will it ever be empty? won't the crd change automagically set the default for us?

Reply:
i did not know that was even a thing 😱